src/routes/safe/components/Settings/SpendingLimit/ResetTime.tsx

```diff
@@ -0,0 +1,113 @@
+import { RadioButtons, Text } from '@gnosis.pm/safe-react-components'
+import { FormControlLabel, hexToRgb, Switch as SwitchMui } from '@material-ui/core'
+import React from 'react'
+import { useField } from 'react-final-form'
+import { Field } from 'src/routes/safe/components/Settings/SpendingLimit/Amount'
+import styled from 'styled-components'
+
+// TODO: propose refactor in safe-react-components based on this requirements
+const SpendingLimitRadioButtons = styled(RadioButtons)`
+  & .MuiRadio-colorPrimary.Mui-checked {
+    color: ${({ theme }) => theme.colors.primary};
+  }
+`
+const StyledSwitch = styled(({ ...rest }) => <SwitchMui {...rest} />)`
+  && {
+    .MuiIconButton-label,
+    .MuiSwitch-colorSecondary {
+      color: ${({ theme }) => theme.colors.icon};
+    }
+
+    .MuiSwitch-colorSecondary.Mui-checked .MuiIconButton-label {
+      color: ${({ theme }) => theme.colors.primary};
+    }
+
+    .MuiSwitch-colorSecondary.Mui-checked:hover {
+      background-color: ${({ theme }) => hexToRgb(`${theme.colors.primary}03`)};
+    }
+
+    .Mui-checked + .MuiSwitch-track {
+      background-color: ${({ theme }) => theme.colors.primaryLight};
+    }
+  }
+`
+
+interface RadioButtonOption {
+  label: string
+  value: string
+}
+
+interface RadioButtonProps {
+  options: RadioButtonOption[]
+  initialValue: string
+  groupName: string
+}
+
+const SafeRadioButtons = ({ options, initialValue, groupName }: RadioButtonProps): React.ReactElement => (
+  <Field name={groupName} initialValue={initialValue}>
+    {({ input: { name, value, onChange } }) => (
+      <SpendingLimitRadioButtons name={name} value={value || initialValue} onRadioChange={onChange} options={options} />
+    )}
+  </Field>
+)
+
+const Switch = ({ label, name }: { label: string; name: string }): React.ReactElement => (
```

Comment:
Can we move this to SRC in the next iteration, we have a checkbox but the radio button is missing.

Reply:
I just added a more detailed TODO for the Switch (that actually exists in the SRC library, but there are missing props that I need to pass to properly integrate it with RFF).

The RadioButton is in the library, but I added some styles here, that we may need to move over too.